ledger/src/ui/src/bootloader.c

```
@@ -0,0 +1,373 @@
+                    clear_pin();
+                    THROW(APDU_OK);
+                    break;
+                case RSK_NEWPIN:
+                    reset_if_starting(RSK_META_CMD_UIOP);
+                    tx = set_pin();
+                    clear_pin();
+                    THROW(APDU_OK);
+                    break;
+                case RSK_ECHO_CMD: // echo
+                    reset_if_starting(RSK_ECHO_CMD);
+                    tx = echo(rx);
+                    THROW(APDU_OK);
+                    break;
+                case RSK_MODE_CMD: // print mode
+                    reset_if_starting(RSK_MODE_CMD);
+                    tx = get_mode();
+                    THROW(APDU_OK);
+                    break;
+                case INS_ATTESTATION:
+                    reset_if_starting(INS_ATTESTATION);
+                    tx = get_attestation(rx, &attestation_ctx);
+                    THROW(APDU_OK);
+                    break;
+                case INS_SIGNER_AUTHORIZATION:
+                    reset_if_starting(INS_SIGNER_AUTHORIZATION);
+                    tx = do_authorize_signer(rx, &sigaut_ctx);
+                    THROW(APDU_OK);
+                    break;
+                case RSK_RETRIES:
+                    reset_if_starting(RSK_RETRIES);
+                    tx = get_retries();
+                    THROW(APDU_OK);
+                    break;
+                case RSK_UNLOCK_CMD: // Unlock
+                    reset_if_starting(RSK_META_CMD_UIOP);
+                    tx = unlock();
+                    // The pin value will also be used in
+                    // BOLOS_UX_CONSENT_APP_ADD command, so we can't wipe the
+                    // pin buffer here
+                    THROW(APDU_OK);
+                    break;
+                case RSK_END_CMD: // return to dashboard
+                    reset_if_starting(RSK_END_CMD);
+                    if (onboarding_mode) {
+                        while (1)
```

Maybe we misunderstood each other when planning this. I thought the idea was to disallow any additional operations from executing after the `RSK_WIPE` command ended. This only hangs the device when the `RSK_END_CMD` command is executed, but nothing prevents a user from executing other commands instead.

---

Hmmm, my bad here. I will change this to hang after `RSK_WIPE`

---

I would try to avoid hanging/leaving the USB unresponsive. Instead, after onboarding I'd just raise an error on any command. How does that sound?